library/src/main/java/com/owncloud/android/lib/resources/shares/OCShare.java

```diff
@@ -222,6 +229,9 @@ public void readFromParcel(Parcel source) {
         isPasswordProtected = source.readInt() == 1;
         hideFileDownload = source.readInt() == 1;
         label = source.readString();
+        hasPreview = source.readInt() == 1;
+        mimetype = source.readString();
+        ownerDisplayName = source.readString();
```

Comment:
Can you create a test for this?
We have already a test here: https://github.com/nextcloud/android-library/blob/master/library/src/androidTest/java/com/owncloud/android/lib/resources/shares/GetSharesRemoteOperationIT.java#L126

Reply:
What would you like me to test? The test you linked only checks share type and path.

Reply:
Correct, there you can add mimetime, hasPreview and ownerDisplayName check.
E.g. in this case it is a folder, with no preview and owner is the user "admin" (I think).
Basically just to check that those attributes are really returned by server and parsed correctly.

Reply:
Done https://github.com/nextcloud/android-library/pull/844/commits/e2baf8d33f5ef6baf212ee75a31e6dfef24bdf00